### PR TITLE
Nix in docker

### PR DESCRIPTION
**Pull request by YPares**
This provides fixes so sparkle build can start in a nix-shell inside a docker container.

The only problem is that it hardcodes STACK_IN_CONTAINER=1; in the shell.nix

### CODE REVIEW

**Comment by robinbb** [comment]
This seems like a good idea. What is the difference in Docker image sizes?

File: shell.nix [hunk 1]
```diff
@@ -28,7 +28,10 @@ let
       optional stdenv.isLinux glibcLocales ++
       [ ghc pkgconfig ];
 
-    STACK_IN_NIX_SHELL=1;
+    SSL_CERT_FILE=/nix-profile/etc/ssl/certs/ca-bundle.crt;
```
Comment (robinbb) [inline shell.nix]: Should this be ~/.nix-profile... ? When building the Dockerfile, I get:

> error: getting attributes of path ‘/nix-profile/etc/ssl/certs/ca-bundle.crt’: No such file or directory
> (use ‘--show-trace’ to show detailed location information)

**Comment by robinbb** [comment]
Does not build for me, as-is. See my comment on the Dockerfile.

**Comment by mboes** [comment]
Fixed by #60.

**Pull request closed without merging**